src/libraries/Lockers.sol

```diff
@@ -94,4 +99,35 @@ library Lockers {
             tstore(slot, count)
         }
     }
+
+    function getCurrentHook() internal view returns (IHooks currentHook) {
+        return IHooks(getHook(length()));
+    }
+
+    function getHook(uint256 i) internal view returns (address hook) {
+        uint256 slot = HOOK_ADDRESS_SLOT + i;
+        assembly {
+            hook := tload(slot)
+        }
+    }
+
+    function setCurrentHook(IHooks currentHook) internal returns (bool set) {
+        // Set the hook address for the current locker if the address is 0.
+        // If the address is nonzero, a hook has already been set for this lock, and is not allowed to be updated or cleared at the end of the call.
+        if (address(getCurrentHook()) == address(0)) {
+            uint256 slot = HOOK_ADDRESS_SLOT + length();
+            assembly {
+                tstore(slot, currentHook)
+            }
+            return true;
+        }
+    }
+
+    // We only want to clear the current hook if the msg.sender is the original locker.
```

Comment:
this comment feels odd because this check doesn't happen here? Is this a warning to devs?

Reply:
Oh yeah old comment I think